### PR TITLE
HDDS-6066. Cleanup OmService injection.

### PR DESCRIPTION
**Pull request by bharatviswa504**
## What changes were proposed in this pull request?

Cleanup OmService Injection. 
With the proposal of HDDS-5881, S3G is not using S3 Tokens for auth, it is using S3G Kerberos credential plus token information is now part of actual Proto OMRequest Object. After this change, we do not need service field which we inject for tokens. 

And also old code has issue, when one of the address is unresolved it fails injection. (Now anyway this code is not needed, we should be good)

## What is the link to the Apache JIRA

[(Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HDDS-XXXX. Fix a typo in YYY.)

Please replace this section with the link to the Apache JIRA)](https://issues.apache.org/jira/browse/HDDS-6066)

## How was this patch tested?

Existing tests. This is mostly removal of not required code.

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java [hunk 1]
```diff
@@ -63,9 +62,6 @@
   @Inject
   private OzoneConfiguration ozoneConfiguration;
 
-  @Inject
-  private Text omService;
-
   @Inject
   private String omServiceID;
```
Comment (kerneltime) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:66]: If we move the initialization of the `omServiceID` to `createOzoneClient` then we can delete `OzoneServiceProvider.java`

Comment (kerneltime) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java:66]: The initialization should still effectively be Application Scoped as the client is initialized only once.

**Comment by kerneltime** [comment]
One minor suggestion.  LGTM

**Comment by kerneltime** [comment]
@bharatviswa504 if you wish to do the further clean up in a separate jira that is fine as well.

**Comment by bharatviswa504** [comment]
> @bharatviswa504 if you wish to do the further clean up in a separate jira that is fine as well.

Thanks, We have a Jira HDDS-4010 which touches the same related code. I will revisit this as part of HDDS-4010 Jira.

**Comment by bharatviswa504** [comment]
@kerneltime 
Thank You for the review.
To fix findbug did few more code changes. Can you take a look at latest changes.

**Comment by kerneltime** [comment]
LGTM. It would be nice to remove `OzoneServiceProvider.java` altogether and introduce it when we have OM federation.

**Comment by hanishakoneru** [comment]
Thanks @bharatviswa504  for working on this and @kerneltime for reviewing it. 
LGTM. +1.

**Comment by bharatviswa504** [comment]
Thank You @hanishakoneru and @kerneltime for the review.

**Pull request merged**